neasqc_wp61/models/quantum/embedder.py

```diff
@@ -0,0 +1,316 @@
+
+                    sentence_word_embeddings.append(word_embedding.tolist())
+                vectorised_sentence_list.append(sentence_word_embeddings)
+
+            embeddings_df["sentence_vectorised"] = vectorised_sentence_list
+            print("Done!")
+
+        self.dataset = embeddings_df
+
+        return self.dataset
+
+    def save_embedding_dataset(self, path: str, filename: str) -> None:
+        """
+        Saves the dataset containing the BERT embeddings as a CSV file.
+
+        Parameters
+        ----------
+        path : str
+            The path to the location where the user wishes to save the
+            generated dataset containing the embeddings.
+        filename: str
+            The name with which the generated dataset containing the
+            embeddings will be saved (it should not include .csv or any
+            other file extension).
+        """
+        self.dataset.to_csv(path + "/" + filename + ".tsv", index=False, sep="\t")
+
+
+class FastText(Embedder):
+    """
+    Class for generating FastText embeddings.
+    """
+
+    def __init__(
+        self,
+        dataset: DataFrame,
+        sentence_embedding: bool = True,
+        dim: int = 300,
+        cased: bool = True,
+    ) -> None:
+        """
+        Initialises the FastText embedder class
+
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            Pandas dataset. Each row of the dataset correspods to a
+            sentence.
+        sentence_embedding : bool
+            States whether we want to produce a sentence embedding or word
+            embedding vector for each sentence in the dataset. True is for
+            sentence embedding, False is for word embedding.
+        dim : int
+            The output dimension of FastText's word/sentence vectors.
+            Default value is 300, but can be set to any value below that.
+        cased: bool
+            Controls whether we casefold our input sentences before
+            vectorising them using FastText. FastText's pretrained model is
+            case sensitive so casefolding will produce different embeddings.
+        """
+        super().__init__(dataset, sentence_embedding)
+        self.dim = dim
+        self.cased = cased
+
+    def compute_embeddings(self) -> DataFrame:
+        """
+        Creates FastText embeddings for sentences in a dataset
+
+        Returns
+        -------
+        embeddings_df : pd.DataFrame
+            A pandas dataframe with the same structure as the original
+            dataset, but with an additional column containing the
+            FastText embeddings corresponding to the sentence contained
+            in each row.
+        """
+        ftu.download_model("en", if_exists="ignore")
+        model = ft.load_model("cc.en.300.bin")
+        if self.dim < 300:
+            ftu.reduce_model(model, self.dim)
+        elif self.dim > 300:
+            print("Error: dimension must be <= 300")
```

Comment:
When you have an error message, you don't just want to print smth. You want to actually throw/raise an error. Pls be mindful of the error type you choose (and the text, but usually the text is straightforward, ppl make more mistakes on error typing)

Reply:
addressed. Thanks Ryk, learning a lot working through these comments!